### PR TITLE
Improve error message handling.

### PR DESCRIPTION
**Pull request by maruel**
Change Transpile() to return an ast.File instead of printing from within the
function.

This is the first part of the big PR.

### CODE REVIEW

File: µ.go [hunk 1]
```diff
@@ -14,17 +14,43 @@
...
+	"go/ast"
+	"io/ioutil"
 	"log"
 	"os"
 
 	"github.com/googlesamples/mugo/transpiler"
 )
 
+func mainImpl() error {
+	verbose := flag.Bool("verbose", false, "log data")
```
Comment (proppy) [inline µ.go:35]: I usually keep the flag as top level unexported global.

File: µ.go [hunk 2]
```diff
@@ -14,17 +14,43 @@
...
+	flag.Parse()
+	if flag.NArg() != 0 {
+		return errors.New("unexpected arguments")
```
Comment (proppy) [inline µ.go:38]: you should probalbly add usage as part of the error.

File: µ.go [hunk 3]
```diff
@@ -14,17 +14,43 @@
...
+	}
+	if !*verbose {
+		log.SetOutput(ioutil.Discard)
```
Comment (proppy) [inline µ.go:41]: maybe we should use glog if we want to control level?

**Comment by proppy** [comment]
Is the intend to avoid logging from the transpile command?

File: transpiler/transpiler.go [hunk 4]
```diff
@@ -29,23 +29,19 @@ import (
 
 // Transpile reads Go source code from the given Reader and writes the
 // transpiled Arduino C++ code to the given Writer.
-func Transpile(out io.Writer, in io.Reader, debug io.Writer) error {
+func Transpile(out io.Writer, in io.Reader) (*ast.File, error) {
```
Comment (proppy) [inline transpiler/transpiler.go:32]: what if instead we passed a ast.File to Transpile?

File: µ.go [hunk 5]
```diff
@@ -14,17 +14,43 @@
...
+	"fmt"
+	"go/ast"
+	"io/ioutil"
 	"log"
 	"os"
 
 	"github.com/googlesamples/mugo/transpiler"
 )
 
+func mainImpl() error {
```
Comment (proppy) [inline µ.go:34]: I think this is small enough it could get merged with `main`, what do you think?

**Comment by maruel** [comment]
See pull #8 for the rationale. May reopen later if I find time. Sorry for the annoyance.

**Pull request closed without merging**